snapcraft/extensions/kde_neon_6.py

```diff
@@ -0,0 +1,272 @@
+                    "target": "$SNAP/qt6",
+                },
+                platform_kf6_snap: {
+                    "content": content_kf6_snap,
+                    "interface": "content",
+                    "default-provider": platform_kf6_snap,
+                    "target": "$SNAP/kf6",
+                },
+            },
+            "environment": {"SNAP_DESKTOP_RUNTIME": "$SNAP/kf6"},
+            "hooks": {
+                "configure": {
+                    "plugs": ["desktop"],
+                    "command-chain": ["snap/command-chain/hooks-configure-desktop"],
+                }
+            },
+            "layout": {"/usr/share/X11": {"symlink": "$SNAP/kf6/usr/share/X11"}},
+        }
+
+    @overrides
+    def get_part_snippet(self, *, plugin_name: str) -> Dict[str, Any]:
+        qt6_sdk_snap = self.kde_snaps.qt6_sdk_snap
+        kf6_sdk_snap = self.kde_snaps.kf6_sdk_snap
+        cmake_args = self.ext_info.cmake_args
+
+        return {
+            "build-environment": [
+                {
+                    "PATH": prepend_to_env(
+                        "PATH",
+                        [
+                            f"/snap/{qt6_sdk_snap}/current/"
+                            f"usr/bin:/snap/{kf6_sdk_snap}/"
+                            f"current/usr/bin"
+                        ],
+                    ),
+                },
+                {
+                    "XDG_DATA_DIRS": prepend_to_env(
+                        "XDG_DATA_DIRS",
+                        [
+                            f"$CRAFT_STAGE/usr/share:"
+                            f"/snap/{qt6_sdk_snap}"
+                            f"/current/usr/share:"
+                            f"/snap/{kf6_sdk_snap}"
+                            f"/current/usr/share",
+                            "/usr/share",
+                        ],
+                    ),
+                },
+                {
+                    "SNAPCRAFT_CMAKE_ARGS": prepend_to_env(
+                        "SNAPCRAFT_CMAKE_ARGS",
+                        [
+                            cmake_args,
+                        ],
+                    ),
+                },
+            ],
+        }
+
+    @overrides
+    def get_parts_snippet(self) -> Dict[str, Any]:
+        # We can change this to the lightweight command-chain when
+        # the content snap includes the desktop-launch from
+        # https://github.com/snapcore/snapcraft-desktop-integration
+        source = get_extensions_data_dir() / "desktop" / "kde-neon-6"
+
+        if self.kde_snaps.qt6_builtin:
+            return {
+                "kde-neon-6/sdk": {
+                    "source": str(source),
+                    "plugin": "make",
+                    "make-parameters": [f"PLATFORM_PLUG={self.kde_snaps.content_kf6}"],
+                    "build-snaps": [
+                        self.kde_snaps.qt6_sdk_snap,
+                        self.kde_snaps.kf6_sdk_snap,
+                    ],
```

Comment:
```suggestion
                    ],
                    "build-packages": [
                        "gettext",
                        "libxml2-utils",
                        "docbook-xml",
                        "docbook-xsl",
                        "libglx-dev",
                        "libgl1-dev",
                        "libglvnd-dev",
                    ],
```

These seem to be bulid requirements for pretty much everything - any reason not to include them always?